### PR TITLE
Tests | Fix TvpTest intermittent failures

### PR DESCRIPTION
**Pull request by DavoudEshtehari**
Extending the PR #1017 approach in similar scenarios.

### CODE REVIEW

File: src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ParameterTest/StreamInputParam.cs [hunk 1]
```diff
@@ -504,17 +503,16 @@ private static void ImmediateCancelText()
                     StringBuilder sb = new StringBuilder();
                     for (int i = 0; i < 1000000; i++)
                         sb.Append(i);
-                    cmd.CommandText = "insert into #blobs (Id, blob) values (1, @blob)";
+                    // Include a delay to allow time for cancellation
+                    cmd.CommandText = "WAITFOR DELAY '00:00:05'; insert into #blobs (Id, blob) values (1, @blob)";
```
Comment (JRahnama) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ParameterTest/StreamInputParam.cs:507]: Curious about the delay time. How did we end up to that exact number of 5 seconds? Is it based on experience or a fact or I am missing something?

Comment (DavoudEshtehari) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ParameterTest/StreamInputParam.cs:507]: It's based on experience; you can find it in another test in the same class that it's already running well.

Comment (cheenamalhotra) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ParameterTest/StreamInputParam.cs:507]: Can we avoid this delay or reduce it to 1-2 secs? It eventually adds up to test execution time 👀

Comment (DavoudEshtehari) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ParameterTest/StreamInputParam.cs:507]: These tests are supposed to be canceled immediately, and this waiting time is just for safety. I'm afraid reducing them may increase the chance of failure.

**Review by JRahnama — APPROVED** [review]

**Pull request merged**